src/widgets/paragraph.rs

```diff
@@ -158,8 +166,15 @@ impl<'a> Widget for Paragraph<'a> {
             )
         });
 
-        let mut line_composer: Box<dyn LineComposer> = if let Some(Wrap { trim }) = self.wrap {
-            Box::new(WordWrapper::new(styled, text_area.width, trim))
+        let mut line_composer: Box<dyn LineComposer> = if let Some(wrap_mode) = self.wrap {
+            match wrap_mode {
+                Wrap::CharBoundary => {
+                    Box::new(CharWrapper::new(styled, text_area.width, self.trim))
+                }
+                Wrap::WordBoundary => {
+                    Box::new(WordWrapper::new(styled, text_area.width, self.trim))
+                }
+            }
```

Comment:
Unfortunately `styled` is of type `Map<Iter<Line>, impl Fn(&Line) -> (FlatMap<Iter<Span>, impl Iterator<Item = StyledGrapheme>, impl Fn(&Span) -> impl Iterator<Item = StyledGrapheme>>, Alignment)>`. I'm not sure it would be cleaner to extract this.

Reply:
Lolwut! :D